lib/services/alert_api/fpas.dart

```
@@ -62,6 +63,15 @@ class FPASApi implements AlertAPI {
       },
     );
 
+    switch (response.statusCode) {
+      case 200: //nothing to do
+        break;
+      case 400:
+        throw InvalidSubscriptionError();
+      default:
```

This seems like a dangerous default. Not being 200 or 400 doesn't mean an unreachable server at all, it could mean all kinds of things. I wouldn't catch that instance and instead only catch the exceptions we know can happen, which right now is just 400.

---

Any other status code than 200 means that the request failed. We could instead throw something like an `undefinedServerError` or how should we handle a broken and unrecognized error in the server connection? To not catch the server error just results in an exception when we try to decode the body with a less helpful exception type, won't it?

---

So we are now throwing an undefinedServerError with the http status code and the response body. In this way, we can handle even unexpected server responses. Or do you have any arguments why we should not throw an exception like this?

---

I don't remember what it was before, but this is fine for me. As long as _something_ is thrown and it isn't specific what it is (because we don't know).